rules-tests/CodeQuality/Rector/Property/DoctrineTargetEntityStringToClassConstantRector/Fixture/Attribute/keep_self_reference.php.inc

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Rector\Doctrine\Tests\CodeQuality\Rector\Property\DoctrineTargetEntityStringToClassConstantRector\Fixture\Attribute;
+
+use Doctrine\ORM\Mapping as ORM;
+
+final class KeepSelfReference
```

Comment:
Do we need the Entity Attribute here?

Reply:
Good point, just added :+1:

Reply:
Checking other fixtures, it's actually not needed. This rules checks directly these annotations/attributes.

There already is another test fixture for `self::class` as well :+1:

Reply:
Maybe add:
`#[ORM\JoinColumn(name: 'some_entity', onDelete: 'SET NULL')`

from my screenshot

Reply:
Ah the difference is, in your existing test you use `private ?self.....`

while here is no type hint.

Please do:
```diff
- private $someEntity;
+ private KeepSelfReference $someEntity;
```

if this does not work, try to make it nullable:

```diff
- private $someEntity;
+ private ?KeepSelfReference $someEntity;
```